packages/flutter_tools/lib/src/base/process_manager.dart

```diff
@@ -479,3 +492,296 @@ class _RecordingProcess implements Process {
   @override
   bool kill([ProcessSignal signal = ProcessSignal.SIGTERM]) => delegate.kill(signal);
 }
+
+/// A [ProcessManager] implementation that mocks out all process invocations
+/// by replaying a previously-recorded series of invocations, throwing an
+/// exception if the requested invocations substantively differ in any way
+/// from those in the recording.
+///
+/// Recordings are expected to be of the form produced by
+/// [RecordingProcessManager]. Namely, this includes:
+///
+/// - a [_kManifestName](manifest file) encoded as UTF-8 JSON that lists all
```

Comment:
In practice we don't actually publish the flutter tools docs anyway. They're for internal use basically. We just use dartdoc format for consistency.